### PR TITLE
Reduce HTTP client creations in the Keystone connector

### PR DESCRIPTION
**Pull request by erwinvaneyk**
#### Overview

Refactors the Keystone connector to reduce the number of instantiations of HTTP clients.

#### What this PR does / why we need it

Improves performance and reduces boilerplate.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by nabokihms** [comment]
@erwinvaneyk I like the idea! Could you also fix the tests so that we can merge this PR?

**Comment by erwinvaneyk** [comment]
Hi @nabokihms thanks for the quick review, will look into the CI failures. They were running fine locally.

File: connector/keystone/keystone.go [hunk 1]
```diff
@@ -111,11 +113,12 @@ var (
...
-		c.Host,
-		c.AdminUsername,
-		c.AdminPassword,
-		logger,
+		Domain:        c.Domain,
+		Host:          c.Host,
+		AdminUsername: c.AdminUsername,
+		AdminPassword: c.AdminPassword,
+		Logger:        logger,
+		client:        http.DefaultClient,
```
Comment (nabokihms) [inline connector/keystone/keystone.go:121]: The test's problem is that they didn't use the `Open` method and manually create connector instances. You should add `http.DefaultClient` to every `conn{}` in tests.

**Review by nabokihms — CHANGES_REQUESTED** [review]
Waiting for tests to be fixed.

**Comment by erwinvaneyk** [comment]
@nabokihms fixed the tests. Thanks for the pointer, my local setup was skipping some of them apparently.

**Review by nabokihms — APPROVED** [review]

**Comment by nabokihms** [comment]
@erwinvaneyk, many thanks. Great work.

**Pull request merged**